libs/market-list/src/lib/utils/market-list.utils.ts

```diff
@@ -0,0 +1,82 @@
+import BigNumber from 'bignumber.js';
+import type {
+  MarketList,
+  MarketList_markets,
+  MarketList_markets_candles,
+} from '../components/__generated__/MarketList';
+
+export const priceChangePercentage = (m: MarketList_markets): BigNumber => {
+  const change = priceChange(m);
+  const { candles } = m;
+  if (change && candles && candles.length > 0) {
+    const yesterdayLastPrice = price(candles[0]?.close, m.decimalPlaces);
+    return change.dividedBy(yesterdayLastPrice).multipliedBy(100);
+  }
+  return new BigNumber(0);
+};
+
+export const priceChange = ({
+  candles,
+  decimalPlaces,
+}: MarketList_markets): BigNumber => {
+  if (candles && candles.length > 0) {
+    const yesterdayLastPrice = price(candles[0]?.close, decimalPlaces);
+    const recentLastPrice = price(
+      candles[candles.length - 1]?.close,
+      decimalPlaces
+    );
+    if (recentLastPrice !== 'N/A' && yesterdayLastPrice !== 'N/A') {
+      return recentLastPrice.minus(yesterdayLastPrice);
+    }
+    return new BigNumber(0);
+  }
+  return new BigNumber(0);
+};
+
+export const candles = ({ candles, decimalPlaces }: MarketList_markets) =>
+  candles?.map((c: MarketList_markets_candles | null) => ({
+    open: price(c?.open, decimalPlaces),
+    close: price(c?.close, decimalPlaces),
+  }));
+
+export const price = (
+  value: number | string | null | undefined,
+  decimalPlaces: number
+): BigNumber | 'N/A' =>
+  value === undefined || value === null
```

Comment:
In other places we display '-' when there is no value, "N/A" can be hard to localize . I would rather return undefined and let function that formats/renders value to decide what to render if there is no value.

Reply:
I don't think we need this function at all.  We should just pass value to component which renders number and use formatNumber (and optinally PriceCell if we want to handle empty values)

Reply:
changing this now.